godot-core/src/obj/script.rs

```diff
@@ -337,6 +342,40 @@ pub unsafe fn create_script_instance<T: ScriptInstance>(
     }
 }
 
+/// Checks if an instance of the script exists for a given object.
+///
+/// This function both checks if the passed script matches the one currently assigned to the passed object, as well as verifies that
+/// there is an instance for the script.
+///
+/// Use this function to implement [`IScriptExtension::instance_has`](crate::classes::IScriptExtension::instance_has).
+#[cfg(since_api = "4.2")]
+pub fn script_instance_exists<O, S>(object: &Gd<O>, script: &Gd<S>) -> bool
+where
+    O: Inherits<Object>,
+    S: Inherits<Script> + IScriptExtension + super::Bounds<Declarer = super::bounds::DeclUser>,
+{
+    let object_script_variant = object.upcast_ref().get_script();
+
+    if object_script_variant.is_nil() {
+        return false;
+    }
+
+    let object_script: Gd<Script> = object_script_variant.to();
+
+    if object_script.upcast_ref::<Script>().__object_ptr() != script.upcast_ref().__object_ptr() {
+        return false;
+    }
+
+    let Some(language) = script.bind().get_language() else {
+        return false;
+    };
+
+    let get_instance_fn = sys::interface_fn!(object_get_script_instance);
+    let instance = unsafe { get_instance_fn(object.obj_sys(), language.obj_sys()) };
+
+    !instance.is_null()
+}
```

Comment:
Maybe in the future we can revisit some parts:
- `__object_ptr()` access
- safety annotation, see [here](https://github.com/godot-rust/gdext/pull/1013/commits/5f98714fa176be7f8730a4da80dd94a95e5e4080#r1923345604)
- `Variant::object_id()`, once polyfill works, see [here](https://github.com/godot-rust/gdext/pull/1013#discussion_r1922832119)

Reply:
Yes, let me know when you go the `Variant::object_id()` polyfill to work, and I can integrate these points.

Reply:
Now implemented in #1034.

For cases where you're sure the object is still alive, there's also the slightly more efficient `object_id_unchecked()`.